contracts/modules/tokenswap/TokenSwapModule.sol

```diff
@@ -190,38 +196,112 @@ contract TokenSwapModule is ModuleBaseWithFee {
             _tokens,
             _pathFrom,
             _pathTo,
+            _daoplomats,
+            _rewardPathTo,
             _deadline
         );
         return lastDealId;
     }
 
+    function _validateCreateSwapInput(
+        address[] memory _daos,
+        address[] memory _tokens,
+        uint256[][] memory _pathFrom,
+        uint256[][] memory _pathTo,
+        address[] memory _daoplomats,
+        uint256[][] memory _rewardPathTo,
+        bytes32 _metadata,
+        uint32 _deadline
+    ) internal view {
+        require(_metadata != "", "TokenSwapModule: Error 101");
+        require(_metadataDoesNotExist(_metadata), "TokenSwapModule: Error 203");
+        require(_daos.length >= 2, "TokenSwapModule: Error 204");
+        require(_tokens.length != 0, "TokenSwapModule: Error 205");
+        require(_deadline != 0, "TokenSwapModule: Error 101");
+        // Check outer arrays
+        uint256 pathFromLen = _pathFrom.length;
+        require(
+            _tokens.length == pathFromLen && pathFromLen == _pathTo.length,
+            "TokenSwapModule: Error 102"
+        );
+        // Check duplicate token addresses
+        for (uint256 i; i < _tokens.length; ++i) {
+            for (uint256 j = i + 1; j < _tokens.length; ++j)
+                require(_tokens[i] != _tokens[j], "TokenSwapModule: Error 104");
+        }
+        // Check inner arrays
+        uint256 daosLen = _daos.length;
+        for (uint256 i; i < pathFromLen; ++i) {
+            require(
+                _pathFrom[i].length == daosLen &&
+                    _pathTo[i].length == daosLen << 2,
+                "TokenSwapModule: Error 102"
+            );
+        }
+        // ToDo: Add comment
+        uint256 daoplomatLen = _daoplomats.length;
+        if (daoplomatLen == 0) {
+            require(
+                daoplomatLen == _rewardPathTo[0].length &&
+                    daoplomatLen == _rewardPathTo[1].length,
+                "TokenSwapModule: Error 102"
+            );
```

Comment:
Does this work? Is `[0]` not the total sum and `[1]` contains an entry for each diplomat? Referring to line 245 here.

Reply:
So I wasn't sure yet if the FE would set the total reward to `0` or leave the array empty. This is written whereby the both arrays would be left empty. Do you think it makes more sense to let them parse a value of `0`?

Reply:
Ah, I see, so that was related to the discussion that we had on Telegram briefly. I still am a bit confused still, since for example:

`daoplomatLen = 3`
`_rewardPathTo[0].length` would then be `1` since it just contains the total percent amount?
`_rewardPathTo[1].length` would indeed be `3`, as there is an entry for each of them. 

What am I missing? 🤔

Reply:
So I changed the requires a bit in comparison of what we see above. Also there is an if statement that makes it so that we only check these requires if `daoplomatLen == 0 || absolute reward == 0`

Reply:
Nice, will check that and comment there if necessary!